config/config.go

```diff
@@ -966,8 +905,18 @@ func (p *PackageConfig) validate(path string) error {
 		return utils.NewConfigValidationError(path, errors.New("empty package id"))
 	}
 
+	if p.Type == "" {
+		// for backwards compatability
+		p.Type = PackageTypeMlModel
+	}
```

Comment:
I assume this is okay because `resource.Config` also modifies the config during validation

Reply:
I mean -- I assume it is okay to mutate the config at this stage in the pipeline. There are other validation spots where modifying the config will cause reconfiguration loops (I discovered this the hard way while on the slam team...)